vivisect/impemu/emulator.py

```diff
@@ -50,14 +49,20 @@ def __init__(self, vw, logwrite=False, logread=False):
 
         self.stack_map_mask = e_bits.sign_extend(0xfff00000, 4, vw.psize)
         self.stack_map_base = e_bits.sign_extend(0xbfb00000, 4, vw.psize)
-        self.stack_pointer = self.stack_map_base + 4096
+        self.stack_pointer = self.stack_map_base + stacksize
 
         # Possibly need an "options" API?
         self._safe_mem = True   # Should we be forgiving about memory accesses?
         self._func_only = True  # is this emulator meant to stay in one function scope?
 
         # Map in a memory map for the stack
-        self.addMemoryMap(self.stack_map_base, 6, "[stack]", init_stack_map)
+        stack_map = init_stack_map
+        if stacksize != 4096:
+            stack_map = ''
+            for i in xrange(stacksize):
```

Comment:
please check this. i'd have thought this should be `for i in xrange(stacksize/4)`, to fill up the buffer with exactly `stacksize` bytes, but this didn't seem to work (the stack appeared to be uninitialized). probably, i just misunderstand the meaning of a few of the variables, so let me know what you think the appropriate value is.